salt/journal-cms/init.sls

```diff
@@ -297,22 +297,41 @@ site-was-installed-check:
             - cmd: site-install
 
 site-install:
-    cmd.run:
-        - name: |
-            set -e
-            ../vendor/bin/drush site-install minimal --existing-config -y
-            ####test -e /home/{{ pillar.elife.deploy_user.username }}/site-was-installed.flag && ../vendor/bin/drush cr || echo "site was not installed before, not rebuilding cache"
-            #../vendor/bin/drush cr # may fail with "You have requested a non-existent service "cache.backend.redis"
-            redis-cli flushall
-        - cwd: /srv/journal-cms/web
-        - runas: {{ pillar.elife.deploy_user.username }}
-        - require:
-            - journal-cms-repository
-        # always perform a new site-install on dev and ci
-        {% if pillar.elife.env not in ['dev', 'ci'] %}
-        - unless:
-            - sudo -u {{ pillar.elife.deploy_user.username}} ../vendor/bin/drush cget system.site name
-        {% endif %}
+  cmd.run:
+    - name: |
+        set -e
+        ../vendor/bin/drush site-install minimal --existing-config -y
```

Comment:
Is it your intention to remove `site-install` step? With this change to the `site-install` step it will reinstall drupal every time `sudo salt-call state.highstate`.

If you do remove `site-install` it is a dependency of a couple of other steps:

- https://github.com/elifesciences/journal-cms-formula/blob/bc4cc5ca933f09504aa6f3de6d3350069ac92e77/salt/journal-cms/init.sls#L342
- https://github.com/elifesciences/journal-cms-formula/blob/bc4cc5ca933f09504aa6f3de6d3350069ac92e77/salt/journal-cms/init.sls#L437

I'm not sure what you will insert as the dependency if `site-install` goes because there will be 3 different step names that could represent this stage depending on the environment:

- `site-install_skip`
- `site-install_dev_ci`
- `site-install_other`